### PR TITLE
Update median feeds after feed_lifetime_sec changed

### PR DESCRIPTION
**Pull request by abitmore**
PR for #890.

TODO:
- [ ] finish test case.

### CODE REVIEW

File: libraries/chain/asset_evaluator.cpp [hunk 1]
```diff
@@ -407,20 +407,33 @@ void_result asset_update_bitasset_evaluator::do_evaluate(const asset_update_bita
...
+      asset_bitasset_data_object& bdo, const asset_object& asset_to_update)
 {
-   // If the minimum number of feeds to calculate a median has changed, we need to recalculate the median
+   const fc::time_point_sec& next_maint_time = db.get_dynamic_global_properties().next_maintenance_time;
+
+   // If the minimum number of feeds to calculate a median has changed
+   // we need to recalculate the median
    bool should_update_feeds = false;
-   if( op.new_options.minimum_feeds != bdo.options.minimum_feeds )
+   if( op.new_options.minimum_feeds != bdo.options.minimum_feeds)
```
Comment (pmconrad) [inline libraries/chain/asset_evaluator.cpp]: Please undo this whitespace change.

File: libraries/chain/asset_evaluator.cpp [hunk 2]
```diff
@@ -407,20 +407,33 @@ void_result asset_update_bitasset_evaluator::do_evaluate(const asset_update_bita
  * @param db the database
  * @param bdo the actual database object
  * @param asset_to_update the asset_object related to this bitasset_data_object
+ * @returns true if the feed price is changed, and after hf 890
  */
-static void update_bitasset_object_options(
+static bool update_bitasset_object_options(
       const asset_update_bitasset_operation& op, database& db,
-      asset_bitasset_data_object& bdo, const asset_object& asset_to_update )
+      asset_bitasset_data_object& bdo, const asset_object& asset_to_update)
```
Comment (pmconrad) [inline libraries/chain/asset_evaluator.cpp]: Please undo this whitespace change.

File: libraries/chain/asset_evaluator.cpp [hunk 3]
```diff
@@ -451,7 +464,16 @@ static void update_bitasset_object_options(
    }
 
    if( should_update_feeds )
+   {
+      const auto old_feed_price = bdo.current_feed.settlement_price;
       bdo.update_median_feeds(db.head_block_time());
+
+      return next_maint_time > HARDFORK_CORE_890_TIME && old_feed_price != bdo.current_feed.settlement_price;
+   }
+   else
```
Comment (pmconrad) [inline libraries/chain/asset_evaluator.cpp]: You can remove the "else" and always return false.

**Comment by abitmore** [comment]
Merged to `890-change-feed-lifetime` branch, will continue there.

**Pull request merged**